reporting_test.py

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/python3
+
+from unittest.mock import patch
```

Comment:
I very rarely mock.

Reply:
As in, you rarely need to, or there's another mechanism you prefer, or you structure your code so you don't need mock-style mechanisms?

Reply:
I try to avoid the need.  I see tests as a second use of your code.  If it's hard to test, it can mean your code isn't as flexible as it ought to be.  So yeah, different code structure often.  Lots of details to discuss in each actual case.